protocol/consistencychecks.go

```diff
@@ -245,7 +245,11 @@ func (cc *ConsistencyChecks) updateTBs(requestType int, msg *Response,
 	case RegistrationType:
 		df := msg.DirectoryResponse.(*DirectoryProof)
 		if df.AP.ProofType() == m.ProofOfAbsence {
-			if err := cc.verifyReturnedPromise(df, key); err != nil {
+			tbKey := key
+			if msg.Error == ReqNameExisted {
+				tbKey = df.TB.Value
+			}
```

Comment:
> We are all use TOFU here, so I'm not sure if returning ReqNameExisted along with CheckPassed makes more sense?

Can you elaborate?  Because, I think that's what we want.  The `CheckPassed` is telling us that the `ReqNameExisted` is valid, and that the server proved it to us.  The client might still need to act on `ReqNameExisted`.